### DISCUSSION TITLE
Apollo Save Tool v1.4.0

### DISCUSSION THREAD

**Discussion by bucanero** [bucanero/apollo-ps4#117] [Announcements]
_in memory of Luna & Leon_ ❤️ 

![luna leon](https://github.com/bucanero/apollo-ps3/raw/master/data/leonluna.png)

### Added

* Network Tools
  * URL downloader tool (download http/https/ftp/ftps links)
  * Simple local Web Server (full access to console drives)
  * Disable Web Browser history
* Hex Editor for save-data files
* On-screen Keyboard (for text input)
* Activate offline accounts with user-defined account IDs (on-screen keyboard)
* Improved internal Web Server (Online DB support)
* User-defined Online DB URL (`Settings`)
* Improved DLC rebuild (read content details from `.pkg` file)
* Explicit firmware check when importing encrypted saves

### Misc

* Updated Apollo patch engine v0.4.1
  * Skip search if the pattern was not found
  * Improve code types 9, B, D
  * Add value subtraction support (BSD)

<hr /><em>This discussion was created from the release <a href='https://github.com/bucanero/apollo-ps4/releases/tag/v1.4.0'>Apollo Save Tool v1.4.0</a>.</em>

**Comment by shangis12** [comment]
Please help me
I have gta v cusa00419 and im trying to decrypt save game to use gta v save editor but when im trying to apply save its says error no cheats found
And i dont know if i have memory.dat(dont kniw what it is)
Please im frustrated

**Comment by joaquincanete** [comment]
God bless you, greats from santa fe argentina, and force for leon and luna, precius ovejeros velgas, cryin whit you, thanks best tool

**Comment by joaquincanete** [comment]
Sorry for my English but I didn't want to stop accepting you in your pain, beautiful pets and friends

**Comment by ian824** [comment]
Why does firmware 5.05 always error when installing?
